node/service/src/lib.rs

```diff
@@ -1208,6 +1284,77 @@ where
 	})
 }
 
+/// Get the workers path by considering the passed-in `workers_path` option, or possible locations
+/// on the filesystem. See `new_full`.
+fn get_workers_paths(
```

Comment:
The function is basically the same code repeated three times. This can be written as a loop over the possible targets.